codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -354,40 +358,7 @@ private static String decodeComponent(String s, int from, int toExcluded, Charse
         return strBuf.toString();
     }
 
-    private static byte decodeHexByte(String s, int pos, int len) {
-        if (pos + 2 >= len) {
-            throw new IllegalArgumentException("unterminated escape sequence at index " + pos + " of: " + s);
-        }
-        int hi = decodeHexNibble(s.charAt(pos + 1));
-        int lo = decodeHexNibble(s.charAt(pos + 2));
-        if (hi == -1 || lo == -1) {
-            throw new IllegalArgumentException(
-                    "invalid escape sequence '" + s.substring(pos, pos + 3) + "' at index " + pos + " of: " + s);
-        }
-        return (byte) ((hi << 4) + lo);
-    }
-
-    /**
-     * Helper to decode half of a hexadecimal number from a string.
-     * @param c The ASCII character of the hexadecimal number to decode.
-     * Must be in the range {@code [0-9a-fA-F]}.
-     * @return The hexadecimal value represented in the ASCII character
-     * given, or {@code -1} if the character is invalid.
-     */
-    private static int decodeHexNibble(final char c) {
```

Comment:
Move `decodeHexNibble` into `ByteBufUtil` and make it `private static` and use it instead of `Character.digit`